src/Underworld.ts

```diff
@@ -3425,7 +3425,7 @@ ${CardUI.cardListToImages(player.stats.longestSpell)}
     const withinDistance: HasSpace[] = [];
     const potentialTargets = this.getPotentialTargets(prediction);
     for (let entity of potentialTargets) {
-      if (math.distance(entity, target) <= distance) {
+      if (math.sqrDistance(entity, target) <= distance * distance) {
```

Comment:
I dunno about this.
I'd like to see some stats on this as an efficiency gain.  I'm not sure that it's worth the performance for the tradeoff of being confusing to read.
Can you do some perf testing on distance vs sqrDistance and specifically on
1 sqrDistance and 1 `distance*distance` vs just `distance()` please?

Reply:
@jdoleary 
Wow, I'm glad you asked for this! I did some performance testing and I'm confused / VERY surprised by the results. I ran some tests where I calculated sqrDistance() and distance() a up to a million times each in a large number of trials.

With few loops and trials, the results were kind've sporadic and crazy. performance.now wasn't precise enough to measure the time spent calculating. With higher trial and loop counts, the results get more consistent, but not very promising overall. In the best test, sqrDistance was only about 11% faster. In the worst, it was 3% slower. In most trials... it was 0.5-2 percent faster.

This indicates two things
1) For a distance optimization like this to be noticeable, we'd have to calculate it thousands and thousands of times to notice a performance hit.
2) The optimization that sqrDistance provides is totally negligible

So I was wrong. My understanding had always been that using the squared versions of these functions was 5 or even 10 times faster, but maybe that's not the case with modern tech. JavaScript has some pretty awesome auto-optimization, so maybe that's what's going on here, and if so, there's no reason to make changes that decreases the readability of the codebase. I think you're right that we should just use distance in these places, and I'll be reverting most if not all of these changes.

Reply:
I think I'm going to remove all of the sqrDist and sqrMag stuff I implemented over the last 2ish days. 1% faster calculations aren't worth the maintainability and readability of the code. I'm very disappointed that it ended up being as unhelpful as it was, when I thought I was making a nice improvement, but I'm simultaneously relieved we don't have to deal with the sqrX stuff or worry about the performance impact of using magnitude/distance calculations here and there 😅

Reply:
No problem! I'm glad that we have some hard data to back up the decision here!